src/main/java/org/nuxeo/ecm/csv/CSVImporterWork.java

```diff
@@ -295,36 +298,58 @@ protected void doImport(CSVParser parser) {
      * @since 6.0
      */
     protected boolean importRecord(CSVRecord record, Map<String, Integer> header) {
-        final String name = record.get(CSV_NAME_COL);
-        final String type = record.get(CSV_TYPE_COL);
+        String name = record.get(CSV_NAME_COL);
         if (StringUtils.isBlank(name)) {
             log.debug("record.isSet=" + record.isSet(CSV_NAME_COL));
             logError(record.getRecordNumber(), "Missing 'name' value", LABEL_CSV_IMPORTER_MISSING_NAME_VALUE);
             return false;
         }
-        if (StringUtils.isBlank(type)) {
-            log.debug("record.isSet=" + record.isSet(CSV_TYPE_COL));
-            logError(record.getRecordNumber(), "Missing 'type' value", LABEL_CSV_IMPORTER_MISSING_TYPE_VALUE);
-            return false;
+
+        Path targetPath = new Path(parentPath).append(name);
+        name = targetPath.lastSegment();
+        String newParentPath = targetPath.removeLastSegments(1).toString();
+        boolean exists = options.getCSVImporterDocumentFactory().exists(session, newParentPath, name);
+
+        DocumentRef docRef = null;
+        String type = null;
+        if (exists) {
+            docRef = new PathRef(targetPath.toString());
+            type = session.getDocument(docRef).getType();
+        } else {
+            if (Boolean.FALSE.equals(isTypeColumnMissing(header))) {
```

Comment:
`if (!isTypeColumnMissing(header)`?

However, I would prefer to read something like:

    if (hasTypeColumn(header)) {
        type = record.get(CSV_TYPE_COL);
    }
    ....